api/apps/api/src/modules/access-control/projects-acl/project-acl.service.ts

```diff
@@ -92,4 +94,113 @@ export class ProjectAclService implements ProjectAccessControl {
       this.canViewSolutionRoles,
     );
   }
+
+  async isOwner(userId: string, projectId: string): Promise<Permit> {
+    const userIsProjectOwner = await this.roles.findOne({
+      where: {
+        projectId,
+        userId,
+        roleName: Roles.project_owner,
+      },
+    });
+    if (!userIsProjectOwner) {
+      return false;
+    }
+    return true;
+  }
+
+  async hasOtherOwner(userId: string, projectId: string): Promise<Permit> {
+    const allOwnersInProject = await this.roles.count({
+      where: {
+        projectId,
+        roleName: Roles.project_owner,
+        userId: Not(userId),
+      },
+    });
+    return allOwnersInProject >= 1;
+  }
+  /**
+   * @debt This module should not involve user details and it should deal with it
+   * using a standalone module that will access the data just to read it. We
+   * have to get back to it once scenarios, organizations and solutions are included
+   * inside the access-module.
+   */
+  async findUsersInProject(
+    projectId: string,
+    userId: string,
+  ): Promise<UserRoleInProjectDto[] | Permit> {
+    if (!(await this.isOwner(userId, projectId))) {
+      return false;
+    }
```

Comment:
I am not a big fan of returning data or an error condition. I would not change it now, because it works and it's clear enough, but I suggest to eventually either switch to using `Either` from `fp-ts` as we do in several other places, or to use exceptions. Either may be preferable.